buildtest/buildsystem/base.py

```diff
@@ -640,3 +644,274 @@ def _build_testcontent(self):
         lines += [self.recipe.get("run")]
 
         return lines
+
+
+class CompilerBuilder(BuilderBase):
+    type = "compiler"
+
+    # Fortran Extensions Links:
+    # https://software.intel.com/content/www/us/en/develop/documentation/fortran-compiler-developer-guide-and-reference/top/compiler-setup/using-the-command-line/understanding-file-extensions.html
+    # Fortran Extensions: http://fortranwiki.org/fortran/show/File+extensions
+    lang_ext_table = {
+        ".c": "C",
```

Comment:
These compiler-specific variables should belong with the appropriate class.

Reply:
The file extension for C, C++, Fortran are compiler agnostic and not belong to any compiler class. You may be getting confused, the compiler class are not C, Fortran, C++ compiler but rather name of the compiler ``GNU``, ``Intel``, ``PGI``, etc.. Each of these compilers come with a C, C++, Fortran wrapper. That's why i kept ``lang_ext_table`` in CompilerBuilder, the main purpose is to use the extensions to detect the programming language (``self.lang``) which is also compiler agnostic.

Reply:
Ah ok. Is this going to be a common practice - to have a huge table of extensions? Can you think of a simpler way to do this, maybe regular expressions that can be shared across functions and classes instead of a hard coded lookup just with the Compiler class?

Reply:
Fortran is the culprit when it comes to many extensions, i converted the if/else into a lookup table which simplified the logic, the fact is you can write any C, C++ program without a file extension and it would still work so buildtest is trying to enforce some rules when user specify source file. 

Eventually i wanted to add ``.cu`` to the lookup table for CUDA language, it seems easier to add key, value so i can map extension to a language name. Another option was to have a separate list per language and then search in list. Either way it doesn't make much difference, I prefer a lookup table. 

```
c_ext = ['.c']
cplus_ext = ['.cpp', '.cc', ...]
fortran_ext = [.'f', ...]
```

Reply:
Makes sense, let’s have this lookup table (since it’s shared between all compilers) stay as is for now, it will be inherited by child classes.